pandas/core/common.py

```diff
@@ -1403,6 +1403,10 @@ def _fill_zeros(result, x, y, name, fill):
                 if np.isinf(fill):
                     np.putmask(result,(signs<0) & mask, -fill)
 
+                if "floordiv" in name:  # (PR 9308)
+                    nan_mask = ((y == 0) & (x == 0)).ravel()
+                    np.putmask(result, nan_mask, np.nan)
+
```

Comment:
@jreback, I just attempted to refactor `_fill_zeros` to remove all the raveling and reshaping (which aren't necessary).  It actually slowed down my test code (shown below) from 42s to 57s.  

According the NumPy docs for `ravel`,  a "copy is made only if needed".  In our case a copy wasn't being made.  And the `np.putmask` operation seemed to take much longer for the original array than the raveled one ‒ the line `np.putmask(result, mask, fill)` (which I didn't modify) had its average runtime increase from 87ms to 548ms.

I'm also adding a comment to explain why we ravel and reshape.

-------------------------

*Test code:*

    def f():
        xx % 0
        xx / 0
        xx // 0
        yy % 0
        yy / 0
        yy // 0
    
    xx = pd.DataFrame(np.random.random((10000, 10000)))
    yy = pd.DataFrame(np.random.random_integers(0,100, size=(10000,10000)))
    for _ in range(5):
        f()